src/robot/libdocpkg/htmlwriter.py

```diff
@@ -46,153 +42,3 @@ def write(self, line):
-            match = headers.match(line.strip())
-            if match:
-                yield match.group(2)
-
-    def _escape_and_encode_targets(self, targets):
-        return NormalizedDict((html_escape(key), self._encode_uri_component(value))
-                              for key, value in targets.items())
-
-    def _encode_uri_component(self, value):
-        # Emulates encodeURIComponent javascript function
-        return quote(value.encode('UTF-8'), safe="-_.!~*'()")
-
-    def html(self, doc, intro=False):
-        doc = self._doc_to_html(doc)
-        if intro:
-            doc = self._header_regexp.sub(r'<h\1 id="\2">\2</h\1>', doc)
-        return self._name_regexp.sub(self._link_keywords, doc)
-
-    def _link_keywords(self, match):
-        name = match.group(1)
-        if name in self._targets:
-            return '<a href="#%s" class="name">%s</a>' % (self._targets[name], name)
-        return '<span class="name">%s</span>' % name
-
-
-class DocToHtml(object):
-
-    def __init__(self, doc_format):
-        self._formatter = self._get_formatter(doc_format)
-
-    def _get_formatter(self, doc_format):
-        try:
-            return {'ROBOT': html_format,
-                    'TEXT': self._format_text,
-                    'HTML': lambda doc: doc,
-                    'REST': self._format_rest}[doc_format]
-        except KeyError:
-            raise DataError("Invalid documentation format '%s'." % doc_format)
-
-    def _format_text(self, doc):
-        return '<p style="white-space: pre-wrap">%s</p>' % html_escape(doc)
-
-    def _format_rest(self, doc):
-        try:
-            from docutils.core import publish_parts
-        except ImportError:
-            raise DataError("reST format requires 'docutils' module to be installed.")
-        parts = publish_parts(doc, writer_name='html',
-                              settings_overrides={'syntax_highlight': 'short'})
-        return parts['html_body']
-
-    def __call__(self, doc):
-        return self._formatter(doc)
-
-
-class HtmlToText(object):
-    html_tags = {
-        'b': '*',
-        'i': '_',
-        'strong': '*',
-        'em': '_',
-        'code': '``',
-        'div.*?': ''
-    }
-    html_chars = {
-        '<br */?>': '\n',
-        '&amp;': '&',
-        '&lt;': '<',
-        '&gt;': '>',
-        '&quot;': '"',
-        '&apos;': "'"
-    }
-
-    def get_shortdoc_from_html(self, doc):
-        match = re.search(r'<p.*?>(.*?)</?p>', doc, re.DOTALL)
-        if match:
-            doc = match.group(1)
-        doc = self.html_to_plain_text(doc)
-        return doc
-
-    def html_to_plain_text(self, doc):
-        for tag, repl in self.html_tags.items():
-            doc = re.sub(r'<%(tag)s>(.*?)</%(tag)s>' % {'tag': tag},
-                         r'%(repl)s\1%(repl)s' % {'repl': repl}, doc,
-                         flags=re.DOTALL)
-        for html, text in self.html_chars.items():
-            doc = re.sub(html, text, doc)
-        return doc
```

Comment:
Moved `DocFormatter `, `DocToHtml ` and `HtmlToText ` to `htmlutils.py` without any change at all!!!

refactored the `JsonConverter ` into the `LibraryDoc` in `model.py`